client/batch.go

```diff
@@ -308,6 +315,29 @@ func (b *Batch) Scan(s, e interface{}, maxRows int64) {
 	b.initResult(1, 0, nil)
 }
 
+// ReverseScan retrieves the rows in reserve order starting at end (exclusive)
+// and finishing at begin(inclusive).
+//
+// A new result will be appended to the batch which will contain up to maxRows
+// rows and Result.Err will indicate success or failure.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (b *Batch) ReverseScan(s, e interface{}, maxRows int64) {
```

Comment:
I think I've confused myself right now. Let me summarize.
whether the left endpoint is included or not doesn't matter. It's all about `EndKey`. I'm assuming included below.
Currently:
* `[a-z)` (right-exclude)
* descriptor lookup: needs `.Next()` because the reverse scan is exclusive

the alternative, including the right endpoint, is awkward because that's not how our key ranges work.

So I think you've indeed implemented the one that makes most sense.